nil/getting-started/working-with-smart-contracts/deploying-a-contract.mdx

```diff
@@ -0,0 +1,22 @@
+# Deploying a contract
+
+## External deployment
+
+An external deployment of a contract is made by a user or another entity outside of the cluster. 
+
+As is the case with all external messages, an external deployment must be paid by the contract at the address to which the call is maid. However, before deployment, this address is unoccupied. To proceed with external deployment, an entity must first send funds to the address where the contract will reside.
+
+:::info
+
+A new address is always calculated in a way that only permits the owner of a specific keypair to deploy a contract. Once funds are sent to an address, it is considered 'occupied', and only the owner of the funds can deploy a contract to this address.
```

Comment:
"Keypair" is too specific here I guess (e.g., the deployed contract could be a contract without `verifyExternal` and key management at all). The address is calculated from the constructor of the contract that will be deployed there, so only that contract with those exact constructor arguments can occupy it.